### PR TITLE
Typescript migration

### LINKED ISSUE #74: Types for smee-client
It would be great to have TypeScript types for `smee-client`.

### PR DESCRIPTION
**Pull request by wolfy1339**
Fixes #74

### CODE REVIEW

**Comment by wolfy1339** [comment]
Not stale
________________________________

**Comment by wolfy1339** [comment]
Not stale

**Comment by wolfy1339** [comment]
Ping @JasonEtco

**Comment by wolfy1339** [comment]
Not stale

**Comment by wolfy1339** [comment]
Still relevant

**Comment by wolfy1339** [comment]
Ping @JasonEtco @tcbyrd

**Comment by wolfy1339** [comment]
Still relevant

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

File: index.ts [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
+import EventSource = require('eventsource')
+import superagent = require('superagent')
+import url = require('url')
+import querystring = require('querystring')
 
 class Client {
-  constructor ({ source, target, logger = console }) {
+  source: string;
+  target: string;
+  logger: Console;
```
[the remark this answers is not in the document]
Comment (wolfy1339) [reply index.ts:10]: Yes, indeed. they should be updated to those ones

File: index.ts [hunk 2]
```diff
@@ -24,7 +35,7 @@ class Client {
 
     delete data.query
 
-    const req = superagent.post(target).send(data.body)
+    const req = superagent.post(url.format(target)).send(data.body)
```
[the remark this answers is not in the document]
Comment (wolfy1339) [reply index.ts:38]: I don't remember exactly.
But removing it causes errors
```
Argument of type 'UrlWithParsedQuery' is not assignable to parameter of type 'string'.
```

File: index.ts [hunk 3]
```diff
@@ -36,7 +47,7 @@ class Client {
       if (err) {
         this.logger.error(err)
       } else {
-        this.logger.info(`${req.method} ${req.url} - ${res.statusCode}`)
+        this.logger.info(`${req.method} ${req.url} - ${res.status}`)
```
[the remark this answers is not in the document]
Comment (wolfy1339) [reply index.ts:50]: Same comment as above.

```
Property 'statusCode' does not exist on type 'Response'.
```

**Comment by gr2m** [comment]
As we don't have good test coverage yet (see #155), did you try it locally if this actually works? 😁

**Review by ivomusk — APPROVED** [review]

**Pull request merged**